app/client/src/widgets/TabsWidget/widget/index.tsx

```diff
@@ -350,31 +357,36 @@ class TabsWidget extends BaseWidget<
     return [];
   };
 
+  setDefaultSelectedTabWidgetId = () => {
+    const visibleTabs = this.getVisibleTabs();
+    // Find the default Tab object
+    const selectedTab = find(visibleTabs, {
+      label: this.props.defaultTab,
+    });
+    // Find the default Tab id
+    const selectedTabWidgetId = selectedTab
+      ? selectedTab.widgetId
+      : visibleTabs.length
+      ? visibleTabs[0].widgetId
+      : undefined; // in case the default tab is deleted
```

Comment:
We could also replace this with
```suggestion
    const selectedTabWidgetId = selectedTab?.widgetId ??  visibleTabs?.[0]?.widgetId;
```

Also please do modify `selectedTabWidgetId` to `defaultTabWidgetId`